lib/front_matter.ts

```diff
@@ -32,20 +32,30 @@ function split(str: string) {
   return { content: str };
 }
 
-function parse(str: string, options?: yaml.LoadOptions) {
+export type ParseResult = Record<string, any> & Partial<{
+  _content: string;
+  title: string;
+  description: string;
+  thumbnail: string;
+  date: any;
+  updated: any;
+  permalink: string;
+}>
+
+function parse(str: string, options: yaml.LoadOptions = {}) {
   if (typeof str !== 'string') throw new TypeError('str is required!');
 
   const splitData = split(str);
   const raw = splitData.data;
 
   if (!raw) return { _content: str };
 
-  let data;
+  let data: ParseResult;
 
   if (splitData.separator.startsWith(';')) {
     data = parseJSON(raw);
   } else {
-    data = parseYAML(raw, options);
+    data = <any>parseYAML(raw, options);
```

Comment:
```suggestion
    data = parseYAML(raw, options) as any;
```

Prefers `as` over generic syntax.

Reply:
done sir